src/main/java/org/jenkinsci/maven/plugins/hpi/TestDependencyMojo.java

```diff
@@ -390,6 +374,105 @@ public void execute() throws MojoExecutionException {
         }
     }
 
+    private void copyTestDependencies(Set<MavenArtifact> mavenArtifacts) throws MojoExecutionException {
+        File testDir = new File(project.getBuild().getTestOutputDirectory(), "test-dependencies");
+        try {
+            Files.createDirectories(testDir.toPath());
+        } catch (IOException e) {
+            throw new MojoExecutionException("Failed to create directories for '" + testDir + "'", e);
+        }
+
+        ProjectBuildingRequest buildingRequest = new DefaultProjectBuildingRequest(session.getProjectBuildingRequest());
+        buildingRequest.setRemoteRepositories(project.getRemoteArtifactRepositories());
+        List<RemoteRepository> remoteRepositories = RepositoryUtils.toRepos(buildingRequest.getRemoteRepositories());
+
+        List<ArtifactRequest> artifactRequests = new ArrayList<>();
+        boolean ignoreWorkspaceRepository = false;
+        for (MavenArtifact mavenArtifact : mavenArtifacts) {
+            if (!mavenArtifact.isPluginBestEffort(getLog())) {
+                continue;
+            }
+
+            String artifactId;
+            try {
+                artifactId = mavenArtifact.getActualArtifactId();
+            } catch (IOException e) {
+                throw new MojoExecutionException("Failed to resolve " + mavenArtifact.getId(), e);
+            }
+            if (artifactId == null) {
+                getLog().debug("Skipping " + artifactId + " with classifier " + mavenArtifact.getClassifier());
```

Comment:
Since `artifactId` is null, I think this will be a constant string.  Optional to report the constant string directly rather than having Java convert the null to the string "null".  This is not a blocking comment, since the code is well behaved as written.
```suggestion
                getLog().debug("Skipping null artifactID with classifier " + mavenArtifact.getClassifier());
```

Reply:
Sure, this code was there before this PR as well:

```
                if (artifactId == null) {
                    getLog().debug("Skipping " + artifactId + " with classifier " + a.getClassifier());
                    continue;
                }
```

To be honest, I think it is pointless and could just be deleted, but I was trying to touch the original code as little as possible. But I accepted your suggestion because it is an improvement over the status quo.